src/main/java/io/stargate/sgv2/jsonapi/service/cqldriver/executor/CollectionSettings.java

```diff
@@ -94,6 +87,76 @@ public boolean equals(Object o) {
     }
   }
 
+  public record VectorConfig(
+      boolean vectorEnabled,
+      int vectorSize,
+      SimilarityFunction similarityFunction,
+      VectorizeConfig vectorizeConfig) {
+
+    public static VectorConfig notEnabledVectorConfig() {
+      return new VectorConfig(false, -1, null, null);
+    }
+
+    public static VectorConfig fromJson(JsonNode jsonNode) {
+      // dimension, similarityFunction, must exist
+      int dimension = jsonNode.get("dimension").asInt();
+      SimilarityFunction similarityFunction =
+          SimilarityFunction.fromString(jsonNode.get("metric").asText());
+
+      VectorizeConfig vectorizeConfig = null;
+      // construct vectorizeConfig
+      if (jsonNode.has("service")) {
```

Comment:
As far as patterns go, there's often no need to call `if (node.has("xxx"))` before `get("xxx")` -- `get()` returns `null` if no such key exists so that's not a problem.

Reply:
Thanks, resolved